v1.5 Editor tools

ZarfaZ opened the pull request:
1) Added new game mode: editor. In Editor mode you can create new cubes, light cubes, change transparency of each cube (incl. lights) and clear field
2) New game mode available as seperate game (new editor from main menu) or as editor tools during regular maze game

Massive refactoring took place, especially for Field class

ZarfaZ committed:
v1.5 editor tools

Major commit

ZarfaZ committed:
v1.5 Editor release

1) Added new game mode: editor. In Editor mode you can create new cubes, light cubes, change transparency of each cube (incl. lights) and clear field
2) New game mode available as seperate game (new editor from main menu) or as editor tools during regular maze game

Massive refactoring took place, especially for Field class

ZarfaZ committed:
Merge pull request #4 from ZarfaZ/Feature_v1.5

Feature v1.5

ZarfaZ committed:
Small bug fixes

andy128k commented on maze4d/maze4d.vcxproj:
```diff
@@ -185,29 +188,37 @@ xcopy /y /d  "$(ProjectDir)*.hlsl" "$(OutDir)"</Command>
     <ClCompile Include="MainUiController.cpp" />
     <ClCompile Include="Maze.cpp" />
     <ClCompile Include="MainMenu.cpp" />
+    <ClCompile Include="MazeField.cpp" />
```
To satisfy CI build you also need to update a list of files in `CMakeLists.txt` and add all new `.cpp` files there.

ZarfaZ replied:
Thanks Andy, added them and resolved some conflicts.

Have never worked with CMake before, so I used CI action as debugger, which lead to some spamming commits-rebuilds, but seems it's fine now

ZarfaZ committed:
Update CMakeLists.txt

ZarfaZ committed:
Merge branch 'main' into main

ZarfaZ committed:
Update Field.h

ZarfaZ committed:
Update RectangleGraphics.h

ZarfaZ committed:
Small fixes

1) Win block script activated
2) Space button now toggle cursor in menu also
3) Editor tools mouse button clicks now does not work while in menu
